src/aiida/transports/plugins/ssh_auto.py

```diff
@@ -0,0 +1,50 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Plugin for transport over SSH (and SFTP for file transfer)."""
+
+import pathlib
+
+import paramiko
+
+from .ssh import SshTransport
+
+__all__ = ('SshAutoTransport',)
+
+
+class SshAutoTransport(SshTransport):
```

Comment:
As mentioned in one of my comments above, I'm wondering if we should name the plugin `SshConfigTransport` (entry point `ssh_config`), since it takes the information from the `~/.ssh/config`.

Reply:
I see your point of it corresponding to the `~/.ssh/config` file, but I don't think it is as obvious to a user. If a user knows nothing about the way these plugins work, but they have the option between `core.ssh` and `core.ssh_auto`, I think it is safe to say that they will pick the latter if they want the easy experience. In contrast, if the options are `core.ssh` and `core.ssh_config` I think they would guess that `core.ssh_config` would require more configuration. That is why I would vote for `core.ssh_auto` and just have the docstring explain that it automatically parses the config file. This also makes it less tied to exactly what it is parsing automatically if in the future we want to add more sources of information.

Reply:
Hmm, fair enough! Happy to keep auto then. :)